### ISSUE TITLE
Modify children in two list instances of a same list, but all the children of 2nd list instances are copied to 1st list instance

### ISSUE THREAD

**Issue by classcwnd** [CESNET/libnetconf#78]
We used the same model of [issue 77](https://github.com/CESNET/libnetconf/issues/77). 
__I think the two issues are similar. maybe fix one will fix the other. :-)__
eospf:ospf 2 and eospf:ospf 3 are two list instances of container eospf:redistribute, and eospf:redistribute is child of list instance eospf:ospf.

* First we create the two list instances.
```
<?xml version="1.0" encoding="UTF-8"?>
<rpc message-id="106"
     xmlns="urn:ietf:params:xml:ns:netconf:base:1.0"
     xmlns:econfig="urn:ietf:params:xml:ns:yang:my-config"
     xmlns:econtext="urn:ietf:params:xml:ns:yang:my-context"
     xmlns:eospf="urn:ietf:params:xml:ns:yang:my-ospf">
  <edit-config>
    <target>
      <running/>
    </target>
    <config xmlns:xc="urn:ietf:params:xml:ns:netconf:base:1.0">
    <econfig:contexts>
      <econtext:context>
        <econtext:context-name>redistribute</econtext:context-name>
        <econtext:router>
          <eospf:ospf>
            <eospf:ospf>1</eospf:ospf>
            <eospf:area>
              <eospf:area>0.0.0.0</eospf:area>
            </eospf:area>
            <eospf:redistribute>
              <eospf:ospf>
                <eospf:proto-id>2</eospf:proto-id>
                <eospf:metric>2</eospf:metric>
              </eospf:ospf>
              <eospf:ospf>
                <eospf:proto-id>3</eospf:proto-id>
                <eospf:inter-area/>
                <eospf:intra-area/>
                <eospf:metric>3</eospf:metric>
                <eospf:route-map>2</eospf:route-map>
                <eospf:tag>2</eospf:tag>
              </eospf:ospf>
            </eospf:redistribute>
          </eospf:ospf>
        </econtext:router>
      </econtext:context>
    </econfig:contexts>
    </config>
  </edit-config>
</rpc>
]]>]]>
```

* Then we modify the children values of both eospf:ospf 2 and eospf:ospf 3 as following: i.e. change value of <eospf:metric> of eospf:ospf 2 to 222, and change value of <eospf:metric> of eospf:ospf 3 to 333.
```
<?xml version="1.0" encoding="UTF-8"?>
<rpc message-id="106"
     xmlns="urn:ietf:params:xml:ns:netconf:base:1.0"
     xmlns:econfig="urn:ietf:params:xml:ns:yang:my-config"
     xmlns:econtext="urn:ietf:params:xml:ns:yang:my-context"
     xmlns:eospf="urn:ietf:params:xml:ns:yang:my-ospf">
  <edit-config>
    <target>
      <running/>
    </target>
    <config xmlns:xc="urn:ietf:params:xml:ns:netconf:base:1.0">
    <econfig:contexts>
      <econtext:context>
        <econtext:context-name>redistribute</econtext:context-name>
        <econtext:router>
          <eospf:ospf>
            <eospf:ospf>1</eospf:ospf>
            <eospf:area>
              <eospf:area>0.0.0.0</eospf:area>
            </eospf:area>
            <eospf:redistribute>
              <eospf:ospf>
                <eospf:proto-id>2</eospf:proto-id>
                <eospf:metric>222</eospf:metric>
              </eospf:ospf>
              <eospf:ospf>
                <eospf:proto-id>3</eospf:proto-id>
                <eospf:inter-area/>
                <eospf:intra-area/>
                <eospf:metric>333</eospf:metric>
                <eospf:route-map>2</eospf:route-map>
                <eospf:tag>2</eospf:tag>
              </eospf:ospf>
            </eospf:redistribute>
          </eospf:ospf>
        </econtext:router>
      </econtext:context>
    </econfig:contexts>
    </config>
  </edit-config>
</rpc>
]]>]]>
```

* It's dramatic of the result that all the children of eospf:ospf 3 are copied to eospf:ospf 2. The result datastore is as following:

```
<?xml version="1.0" encoding="UTF-8"?>
<datastores xmlns="urn:cesnet:tmc:datastores:file">
  <running lock="">
    <econfig:contexts xmlns:econfig="urn:ietf:params:xml:ns:yang:my-config" xmlns:econtext="urn:ietf:params:xml:ns:yang:my-context" xmlns:eospf="urn:ietf:params:xml:ns:yang:my-ospf">
      <econtext:context>
        <econtext:context-name>redistribute</econtext:context-name>
        <econtext:router>
          <eospf:ospf>
            <eospf:ospf>1</eospf:ospf>
            <eospf:area>
              <eospf:area>0.0.0.0</eospf:area>
            </eospf:area>
            </eospf:area>
            <eospf:redistribute>
              <eospf:ospf>
                <eospf:proto-id>2</eospf:proto-id>
                <eospf:metric>333</eospf:metric>
                <eospf:inter-area xmlns:eospf="urn:ietf:params:xml:ns:yang:my-ospf"/>
                <eospf:intra-area xmlns:eospf="urn:ietf:params:xml:ns:yang:my-ospf"/>
                <eospf:route-map xmlns:eospf="urn:ietf:params:xml:ns:yang:my-ospf">2</eospf:route-map>
                <eospf:tag xmlns:eospf="urn:ietf:params:xml:ns:yang:my-ospf">2</eospf:tag>
              </eospf:ospf>
              <eospf:ospf>
                <eospf:proto-id>3</eospf:proto-id>
                <eospf:inter-area/>
                <eospf:intra-area/>
                <eospf:metric>333</eospf:metric>
                <eospf:route-map>2</eospf:route-map>
                <eospf:tag>2</eospf:tag>
              </eospf:ospf>
            </eospf:redistribute>
          </eospf:ospf>
        </econtext:router>
      </econtext:context>
    </econfig:contexts>
  </running>
  <startup lock=""/>
  <candidate modified="false" lock=""/>
</datastores>
```

* I dumpped out old doc and new doc before create diff_tree in transapi_running_changed() for your reference.

```
netopeer-server[5205]: Merging the node contexts (src/datastore/edit_config.c:2312)
netopeer-server[5205]: Merging the node context (src/datastore/edit_config.c:2359)
netopeer-server[5205]: Merging the node router (src/datastore/edit_config.c:2243)
netopeer-server[5205]: Merging the node ospf (src/datastore/edit_config.c:2243)
netopeer-server[5205]: Merging the node area (src/datastore/edit_config.c:2243)
netopeer-server[5205]: Merging the node redistribute (src/datastore/edit_config.c:2243)
netopeer-server[5205]: Merging the node ospf (src/datastore/edit_config.c:2243)
netopeer-server[5205]: Merging the node metric (src/datastore/edit_config.c:2243)
netopeer-server[5205]: Merging the node text (src/datastore/edit_config.c:2121)
netopeer-server[5205]: Merging the node metric (src/datastore/edit_config.c:2243)
netopeer-server[5205]: Merging the node text (src/datastore/edit_config.c:2121)
netopeer-server[5205]: Merging the node ospf (src/datastore/edit_config.c:2243)
netopeer-server[5205]: Adding a missing node inter-area while merging (src/datastore/edit_config.c:2225)
netopeer-server[5205]: Creating the node inter-area (src/datastore/edit_config.c:1498)
netopeer-server[5205]: Deleting the node inter-area (src/datastore/edit_config.c:1023)
netopeer-server[5205]: Adding a missing node intra-area while merging (src/datastore/edit_config.c:2225)
netopeer-server[5205]: Creating the node intra-area (src/datastore/edit_config.c:1498)
netopeer-server[5205]: Deleting the node intra-area (src/datastore/edit_config.c:1023)
netopeer-server[5205]: Merging the node metric (src/datastore/edit_config.c:2243)
netopeer-server[5205]: Merging the node text (src/datastore/edit_config.c:2121)
netopeer-server[5205]: Adding a missing node route-map while merging (src/datastore/edit_config.c:2225)
netopeer-server[5205]: Creating the node route-map (src/datastore/edit_config.c:1498)
netopeer-server[5205]: Deleting the node route-map (src/datastore/edit_config.c:1023)
netopeer-server[5205]: Adding a missing node tag while merging (src/datastore/edit_config.c:2225)
netopeer-server[5205]: Creating the node tag (src/datastore/edit_config.c:1498)
netopeer-server[5205]: Deleting the node tag (src/datastore/edit_config.c:1023)
netopeer-server[5205]: Merging the node metric (src/datastore/edit_config.c:2243)
netopeer-server[5205]: Merging the node text (src/datastore/edit_config.c:2121)
netopeer-server[5205]: Deleting the node contexts (src/datastore/edit_config.c:1023)
netopeer-server[5205]: RelaxNG validation on subdatastore 1681692778
netopeer-server[5205]: Schematron validation on subdatastore 1681692778
netopeer-server[5205]: !!!!start to dump old_doc
<?xml version="1.0"?>
<econfig:contexts xmlns:econfig="urn:ietf:params:xml:ns:yang:my-config" xmlns:econtext="urn:ietf:params:xml:ns:yang:my-context" xmlns:eospf="urn:ietf:params:xml:ns:yang:my-ospf" xmlns:wd="urn:ietf:params:xml:ns:netconf:default:1.0"><econtext:context><econtext:context-name>redistribute</econtext:context-name><econtext:router><eospf:ospf><eospf:ospf>1</eospf:ospf><eospf:area><eospf:area>0.0.0.0</eospf:area></eospf:area><eospf:redistribute><eospf:ospf><eospf:proto-id>2</eospf:proto-id><eospf:metric>2</eospf:metric></eospf:ospf><eospf:ospf><eospf:proto-id>3</eospf:proto-id><eospf:inter-area/><eospf:intra-area/><eospf:metric>3</eospf:metric><eospf:route-map>2</eospf:route-map><eospf:tag>2</eospf:tag></eospf:ospf></eospf:redistribute><eospf:spf-timers><eospf:ospf-spfdelayspec wd:default="true">5</eospf:ospf-spfdelayspec><eospf:ospf-spfholdtime wd:default="true">10</eospf:ospf-spfholdtime></eospf:spf-timers><eospf:timers><eospf:rib-eof-delay wd:default="true">40</eospf:rib-eof-delay></eospf:timers><eospf:maximum><eospf:redistribute-quantum wd:default="true">2000</eospf:redistribute-quantum></eospf:maximum></eospf:ospf></econtext:router></econtext:context></econfig:contexts>
netopeer-server[5205]: !!!!end to dump old_doc
netopeer-server[5205]: !!!!start to dump new_doc
<?xml version="1.0"?>
<econfig:contexts xmlns:econfig="urn:ietf:params:xml:ns:yang:my-config" xmlns:econtext="urn:ietf:params:xml:ns:yang:my-context" xmlns:eospf="urn:ietf:params:xml:ns:yang:my-ospf" xmlns:wd="urn:ietf:params:xml:ns:netconf:default:1.0"><econtext:context><econtext:context-name>redistribute</econtext:context-name><econtext:router><eospf:ospf><eospf:ospf>1</eospf:ospf><eospf:area><eospf:area>0.0.0.0</eospf:area></eospf:area><eospf:redistribute><eospf:ospf><eospf:proto-id>2</eospf:proto-id><eospf:metric>333</eospf:metric><eospf:inter-area/><eospf:intra-area/><eospf:route-map>2</eospf:route-map><eospf:tag>2</eospf:tag></eospf:ospf><eospf:ospf><eospf:proto-id>3</eospf:proto-id><eospf:inter-area/><eospf:intra-area/><eospf:metric>333</eospf:metric><eospf:route-map>2</eospf:route-map><eospf:tag>2</eospf:tag></eospf:ospf></eospf:redistribute><eospf:spf-timers><eospf:ospf-spfdelayspec wd:default="true">5</eospf:ospf-spfdelayspec><eospf:ospf-spfholdtime wd:default="true">10</eospf:ospf-spfholdtime></eospf:spf-timers><eospf:timers><eospf:rib-eof-delay wd:default="true">40</eospf:rib-eof-delay></eospf:timers><eospf:maximum><eospf:redistribute-quantum wd:default="true">2000</eospf:redistribute-quantum></eospf:maximum></eospf:ospf></econtext:router></econtext:context></econfig:contexts>
netopeer-server[5205]: !!!!end to dump new_doc
```

YANG Model:
```
        <?xml version="1.0" encoding="UTF-8"?>
        <rpc message-id="106"
             xmlns="urn:ietf:params:xml:ns:netconf:base:1.0"
             xmlns:econfig="urn:ietf:params:xml:ns:yang:my-config"
             xmlns:econtext="urn:ietf:params:xml:ns:yang:my-context"
             xmlns:eospf="urn:ietf:params:xml:ns:yang:my-ospf">
          <edit-config>
            <target> <running/> </target>
            <default-operation>none</default-operation>
            <config xmlns:xc="urn:ietf:params:xml:ns:netconf:base:1.0">
Container     <econfig:contexts>   
List           <econtext:context>
Leaf key         <econtext:context-name>redistribute</econtext:context-name>
Container         <econtext:router>
List                <eospf:ospf>
Leaf key             <eospf:ospf>1</eospf:ospf>
List                  <eospf:area>
Leaf key              <eospf:area>0.0.0.0</eospf:area>
                      </eospf:area>
Container             <eospf:redistribute>
List                    <eospf:ospf xc:operation="remove">  
Leaf key                  <eospf:proto-id>2</eospf:proto-id>
                        </eospf:ospf>
                        <eospf:ospf>
                          <eospf:proto-id>3</eospf:proto-id>
                        </eospf:ospf>
                      </eospf:redistribute>
                    </eospf:ospf>
                  </econtext:router>
                </econtext:context>
              </econfig:contexts>
            </config>
          </edit-config>
        </rpc>
        ]]>]]>
```

**Comment by michalvasko** [comment]
Fixed as part of #77, you can try it now.

**Comment by classcwnd** [comment]
Hi 
I've verified the fix. It works now. Thank you for your quick response!
